### PR TITLE
Remove testing database connection URL literals all over the codebase

### LINKED ISSUE #1496: Remove magical database connection URL literals
Literals like "mysql://root:root@tcp/?maxAllowedPacket=0" are everywhere in the codebase.  Remove them after the fix of https://github.com/sql-machine-learning/sqlflow/issues/1495.

Screen Shot 2019-12-19 at 1 04 28 PM

### LINKED ISSUE #1519: Duplicate mock session code
https://github.com/sql-machine-learning/sqlflow/blob/253578d4d2cb4e96f2162d77b97a31c9a0f6af81/pkg/sql/ir/ir_to_proto_test.go#L25-L36

https://github.com/sql-machine-learning/sqlflow/blob/253578d4d2cb4e96f2162d77b97a31c9a0f6af81/pkg/sql/codegen/tensorflow/codegen_test.go#L25-L42

https://github.com/sql-machine-learning/sqlflow/blob/253578d4d2cb4e96f2162d77b97a31c9a0f6af81/pkg/sql/codegen/xgboost/codegen_test.go#L30-L46

### PR DESCRIPTION
**Pull request by wangkuiyi**
Fix https://github.com/sql-machine-learning/sqlflow/issues/1496

Fix https://github.com/sql-machine-learning/sqlflow/issues/1519

### CODE REVIEW

**Comment by wangkuiyi** [comment]
The unit test stage with `SQLFLOW_TEST_DB=hive` failed with the following messages.

```
=== RUN   TestSQL
--- FAIL: TestSQL (10.06s)
    sqlflowserver_test.go:124: 
        	Error Trace:	sqlflowserver_test.go:124
        	Error:      	Received unexpected error:
        	            	rpc error: code = DeadlineExceeded desc = context deadline exceeded
        	Test:       	TestSQL
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0xee3ca9]
goroutine 27 [running]:
testing.tRunner.func1(0xc0000e0c00)
	/usr/local/go/src/testing/testing.go:874 +0x3a3
panic(0xffc780, 0x1d69a50)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
sqlflow.org/sqlflow/pkg/server.TestSQL(0xc0000e0c00)
	/root/go/src/sqlflow.org/sqlflow/pkg/server/sqlflowserver_test.go:125 +0x329
testing.tRunner(0xc0000e0c00, 0x11f48a0)
	/usr/local/go/src/testing/testing.go:909 +0xc9
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:960 +0x350
FAIL	sqlflow.org/sqlflow/pkg/server	10.094s
```

File: pkg/database/testing.go [hunk 1]
```diff
@@ -154,3 +154,22 @@ func GetSessionFromTestingDB() *proto.Session {
 		HdfsUser:     "",
 		HdfsPass:     ""}
 }
+
+// MockURL returns a valid database connection URL.
+func MockURL() string {
```
Comment (typhoonzero) [inline pkg/database/testing.go:159]: Why need a `MockURL` function, when using it, we may not know what database type we are connecting to.

**Review by typhoonzero — APPROVED** [review]
LGTM with a tiny comment

File: pkg/database/testing.go (hunk 1 above)
Comment (wangkuiyi) [reply pkg/database/testing.go:159]: When I go through the codebase, I noticed some unit tests that just need a mockup URL. So I simply added the function database.MockURL().  You are right - we should check if we can remove it later after some code changes.

**Comment by wangkuiyi** [comment]
Fixed the above CI failure. But got a new one.

```
=== RUN   TestGoroutineLeaky
--- FAIL: TestGoroutineLeaky (10.05s)
    leaktest.go:132: leaktest: timed out checking goroutines
    leaktest.go:150: leaktest: leaked goroutine: goroutine 655 [select]:
        database/sql.(*DB).connectionOpener(0xc0009b0000, 0x1322660, 0xc0004fbb00)
        	/usr/local/go/src/database/sql/sql.go:1052 +0xe8
        created by database/sql.OpenDB
        	/usr/local/go/src/database/sql/sql.go:722 +0x15d
    leaktest.go:150: leaktest: leaked goroutine: goroutine 656 [select]:
        database/sql.(*DB).connectionResetter(0xc0009b0000, 0x1322660, 0xc0004fbb00)
        	/usr/local/go/src/database/sql/sql.go:1065 +0xfb
        created by database/sql.OpenDB
        	/usr/local/go/src/database/sql/sql.go:723 +0x193
    leaktest.go:150: leaktest: leaked goroutine: goroutine 657 [select]:
        github.com/go-sql-driver/mysql.(*mysqlConn).startWatcher.func1(0xc00038e2a0, 0xc0009b00c0, 0xc0002adf80)
        	/root/go/pkg/mod/github.com/go-sql-driver/mysql@v1.4.1/connection_go18.go:178 +0xbf
        created by github.com/go-sql-driver/mysql.(*mysqlConn).startWatcher
        	/root/go/pkg/mod/github.com/go-sql-driver/mysql@v1.4.1/connection_go18.go:175 +0xbe
FAIL
coverage: 67.2% of statements
FAIL	sqlflow.org/sqlflow/pkg/server	10.082s
```

**Comment by wangkuiyi** [comment]
This PR is too big to debug and review, thus out-of-date. I am going to separate it into smaller PRs.

**Pull request closed without merging**